commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java

```diff
@@ -0,0 +1,251 @@
+
+        ctx.client.apply(transform);
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        Operations ops = new Operations(ctx.client);
+
+        Address transatcionsAddress = Address.subsystem("transactions");
+
+        Batch batch = new Batch();
+
+        if (nodeIdentifier != null) {
+            batch.writeAttribute(transatcionsAddress, "node-identifier", nodeIdentifier);
+        }
+        if (enableTtsmStatus != null) {
+            batch.writeAttribute(transatcionsAddress, "enable-tsm-status", enableTtsmStatus);
+        }
+        if (timeout != null) {
+            batch.writeAttribute(transatcionsAddress, "default-timeout", timeout);
+        }
+        if (statisticsEnabled != null) {
+            batch.writeAttribute(transatcionsAddress, "statistics-enabled", statisticsEnabled);
+        }
+        if (jts != null) {
+            batch.writeAttribute(transatcionsAddress, "jts", jts);
+        }
+        if (useJournalStore != null) {
+            batch.writeAttribute(transatcionsAddress, "use-journal-store", useJournalStore);
+        }
+        if (journalStoreEnableAsyncIO != null) {
+            batch.writeAttribute(transatcionsAddress, "journal-store-enable-async-io", journalStoreEnableAsyncIO);
+        }
+        if (processIdUuid != null || processIdSocketBinding != null) {
+            if (processIdUuid != null) {
+                batch.writeAttribute(transatcionsAddress, "process-id-uuid", processIdUuid);
+            } else {
+                // process id sock. binding is not allowed along with enabled uuid therefore it needs to be disabled
+                batch.undefineAttribute(transatcionsAddress, "process-id-uuid");
+                if (processIdSocketBinding != null) {
+                    batch.writeAttribute(transatcionsAddress, "process-id-socket-binding", processIdSocketBinding);
+                }
+                if (processIdSocketMaxPorts != null) {
+                    batch.writeAttribute(transatcionsAddress, "process-id-socket-max-ports", processIdSocketMaxPorts);
+                }
+            }
+        }
+        if (socketBinding != null) {
+            batch.writeAttribute(transatcionsAddress, "socket-binding", socketBinding);
+        }
+        if (statusSocketBinding != null) {
+            batch.writeAttribute(transatcionsAddress, "status-socket-binding", statusSocketBinding);
+        }
+        if (recoveryListener != null) {
+            batch.writeAttribute(transatcionsAddress, "recovery-listener", recoveryListener);
+        }
+        if (objectStorePath != null) {
+            batch.writeAttribute(transatcionsAddress, "object-store-path", objectStorePath);
+        }
+        if (objectStoreRelativeTo != null) {
+            batch.writeAttribute(transatcionsAddress, "object-store-relative-to", objectStoreRelativeTo);
+        }
+
+        ops.batch(batch);
+
+    }
+
+    public static final class Builder<THIS extends Builder> {
```

Comment:
Do you expect to have subclasses? If not, the `THIS` machinery is useless (and a bit confusing).